src/reducers/ByProgramReducer.js

```diff
@@ -121,6 +131,17 @@ export const byProgramReducer = (state, action) => {
         isModalOpen: false,
         complete: false,
       };
+    case actions.SELECT_CUSTOMER:
+      return {
+        ...state,
+        customer: value,
+        program: null,
+        orderType: null,
+        period: null,
+        name: '',
+        isModalOpen: false,
+        complete: false,
+      };
```

Comment:
Just a case in the program reducer which, when selecting a customer, sets everything else to null